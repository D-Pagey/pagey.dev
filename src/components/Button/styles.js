import styled from 'styled-components';

export const Button = styled.button.attrs({
    type: 'button'
})`
    background-color: skyblue;
    border: 0;
    cursor: pointer;
    padding: 0.5rem;
`;
