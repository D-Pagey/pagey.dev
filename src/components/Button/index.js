import React from 'react';
import { oneOf, string } from 'prop-types';
import * as S from './styles';

const Button = ({ children, variant }) => {
    switch (variant) {
        default:
            return <S.Button>{children}</S.Button>;
    }
};

Button.propTypes = {
    children: string.isRequired,
    variant: oneOf([''])
};

Button.defaultProps = {
    variant: ''
};

export default Button;
