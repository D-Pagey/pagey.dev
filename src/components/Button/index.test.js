import React from 'react';
import Button from '.';

describe('Button component', () => {
  it('should render', () => {
    const { container } = render(<Button>Click Me</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
