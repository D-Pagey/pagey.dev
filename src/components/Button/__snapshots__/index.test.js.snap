// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render 1`] = `
.c0 {
  background-color: skyblue;
  border: 0;
  cursor: pointer;
  padding: 0.5rem;
}

<button
  class="c0"
  type="button"
>
  Click Me
</button>
`;
